clang-tools-extra/clang-tidy/misc/ConstCorrectnessCheck.cpp

```diff
@@ -89,6 +89,8 @@ void ConstCorrectnessCheck::registerMatchers(MatchFinder *Finder) {
   const auto ConstReference = hasType(references(isConstQualified()));
```

Comment:
On line 86 it appears the author intended to handle "const arrays" in the "pointee check". They are indeed handled correctly if one of the options is true (I don't recall which one). Is there a bug in that check that we should fix instead?

Reply:
Alternatively we might want to correct/remove the comment

Reply:
I think the comment means check the constness of pointer and array instead of const pointer and const array

Reply:
All the tests (including the FP at hand) pass if you remove `unless(arrayType()` from line 87 (as well as the part of the comment), so I think it would be preferable to do that instead of creating a `ConstArrayType`.